docs/pages/debugging/creating-devtools-plugins.mdx

```diff
@@ -0,0 +1,109 @@
+---
+title: Creating Dev Tools Plugins
+description: Learn how to make your own dev tools plugin to enhance your development experience.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+
+> Dev Tools plugins are compatible with SDK 50 and above. The process for creating a plugin is still in beta and may change in the future.
+
+You can make your own dev tools plugin, whether that's for inspecting aspects of a common framework or library, or something specific to your custom code.
+
+A plugin consists of an Expo app for displaying the dev tool's web user interface, an **expo-module.config.json** file to help the Expo CLI detect the plugin, and then calls to `expo/devtools` API to help your app communicate back and forth with the dev tool's web UI. Plugins can be included inside your app's monorepo or distributed as a separate package on NPM.
+
+## Creating a plugin
+
+### Create an Expo app
+
+The web interface for your dev tool will be an Expo app. Create a new blank Expo app:
+
+<Terminal cmd={['$ npx create-expo-app']} />
+
+Then add the packages needed to build the app for web and to use the dev tools API:
+
+<Terminal cmd={['$ npx expo install react-native-web@~0.19.6 react-dom@18.2.0 @expo/webpack-config']} />
+
+### Update package.json
+
+Move any dependencies from `dependencies` to `devDependences` in **package.json**. These dependencies are only needed for building the web version of the dev tools UI, and putting them in `devDependencies` will ensure they are not included in your app.
+
+```diff package.json
+"dependencies": {
+-  "expo": "~50.0.0",
+-  "react": "18.2.0",
+-  "react-dom": "18.2.0",
+-  "react-native": "0.73.0",
+-  "react-native-web": "~0.19.6"
+},
+"devDependencies": {
+  "@babel/core": "^7.20.0",
+  "@expo/webpack-config": "^18.1.2",
++  "expo": "~50.0.0",
++  "react": "18.2.0",
++  "react-dom": "18.2.0",
++  "react-native": "0.73.0",
++  "react-native-web": "~0.19.6"
+},
+```
+
+Add a `homepage` field to define the exported base URL for web in the format of `/_expo/plugins/{pluginName}`. For example:
+
+```diff package.json
+  "name": "expo-plugin-helloworld",
+  "version": "1.0.0",
++  "homepage": "/_expo/plugins/expo-plugin-helloworld",
```

Comment:
this is for webpack
for metro bundler, we use the `baseUrl` instead, e.g.: https://github.com/expo/dev-plugins/blob/7b8e083ae220941bd7ff84eee1fed2fcd4eb6db5/packages/react-navigation/webui/app.json#L6-L8